cmd/juju/storage/filesystemlistformatters.go

```diff
@@ -42,30 +42,53 @@ func formatFilesystemListTabular(writer io.Writer, infos map[string]FilesystemIn
 		for machineId, machineInfo := range info.Attachments.Machines {
 			filesystemAttachmentInfo := filesystemAttachmentInfo
 			filesystemAttachmentInfo.MachineId = machineId
-			filesystemAttachmentInfo.MachineFilesystemAttachment = machineInfo
+			filesystemAttachmentInfo.FilesystemAttachment = machineInfo
 			for unitId, unitInfo := range info.Attachments.Units {
 				if unitInfo.MachineId == machineId {
 					filesystemAttachmentInfo.UnitId = unitId
 					filesystemAttachmentInfo.UnitStorageAttachment = unitInfo
 					break
 				}
 			}
+			haveMachines = true
+			filesystemAttachmentInfos = append(filesystemAttachmentInfos, filesystemAttachmentInfo)
+		}
+
+		for _, machineInfo := range info.Attachments.Containers {
+			filesystemAttachmentInfo := filesystemAttachmentInfo
+			filesystemAttachmentInfo.FilesystemAttachment = machineInfo
+			// TODO(caas) - fill out details
 			filesystemAttachmentInfos = append(filesystemAttachmentInfos, filesystemAttachmentInfo)
 		}
 	}
 	sort.Sort(filesystemAttachmentInfos)
 
+	if haveMachines {
```

Comment:
This is cunning but it might be better to just have two different output functions, one for the machine case and one for the unit case.

Reply:
99% of the output is the same - just one column is missing, so it would be a lot of duplication.
More needs to be done here as storage progresses so I can revisit as stuff gets added.